core/domain/entities/route_match/specifications/admin/WordPressPostsListTable.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace EventEspresso\core\domain\entities\route_match\specifications\admin;
+
+use EventEspresso\core\domain\entities\route_match\RouteMatchSpecification;
+
+/**
+ * Class WordPressPostsListTable
+ * Returns true when the current request is for the WordPress Posts List table admin page
+ *
+ * @package EventEspresso\core\domain\entities\route_match\specifications
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class WordPressPostsListTable extends RouteMatchSpecification
+{
+    /**
+     * returns true if current request matches specification
+     *
+     * @since $VID:$
+     * @return boolean
+     */
+    public function routeMatches()
+    {
+        return strpos($this->request->requestUri(),'wp-admin/edit.php' ) !== false
+            && (
+                $this->request->getRequestParam('post_type') === 'post'
+                || $this->request->requestParamIsSet('post_type') === false
```

Comment:
In native wp posts lists table loading, it appears the page will invoke `wp_die` if either there is no `post_type` set in the request or the value is not in the array returned by `get_post_types( 'show_ui' => true )`.  I realize that this specification check is likely invoked before the native wp post list table check but I wonder if it'd be more accurate to remove this second `|| $this->request->requestParamIsSet( 'post_type' ) === false` check?

Reply:
> In native wp posts lists table loading, it appears the page will invoke wp_die if either there is no post_type set in the request or the value is not in the array returned by get_post_types( 'show_ui' => true ).

mmmm... not seeing that in `WP_Posts_List_Table` nor in `WP_List_Table`, where is that happening?
 
`WP_Posts_List_Table` gets the post_type from the current `WP_Screen`, which we could do if we loaded the `RouteMatchSpecification` classes much later in the request, but that would obviously make them less usable for making loading decisions earlier on. It's possible we could work the use of `WP_Screen` into them via another method that can only be called after a certain hook point has run. Other implementers of `WP_List_Table`, such as `WP_Comments_List_Table` neither require `post_type` to be set as a request parameter nor as a property on `WP_Screen` (ie: the comments list table for standard WP Posts will not have `post_type` set at all)

>  I wonder if it'd be more accurate to remove this second || $this->request->requestParamIsSet( 'post_type' ) === false check?

Well in some cases the full url for the WP Posts list table is simply `http://local.wordpress.test/wp-admin/edit.php` with no post type specified, but other times, depending on where you navigate from (such as by using the `All`, `Published`, `Draft` links in the table header), it will be `http://local.wordpress.test/wp-admin/edit.php?post_type=post`.  Other post types appear to always display the `post_type` in the url, which subsequently gets added to the `WP_Screen` object. 

What we could do, is utilize the second parameter for the `RequestInterface::getRequestParam()` method which allows you to set a default value **if** the parameter is not set in the request parameters. So instead of:

```php
        return strpos($this->request->requestUri(),'wp-admin/edit.php' ) !== false
            && (
                $this->request->getRequestParam('post_type') === 'post'
                || $this->request->requestParamIsSet('post_type') === false
            );
```

we could simply do something like:

```php
        return strpos($this->request->requestUri(),'wp-admin/edit.php' ) !== false
            && $this->request->getRequestParam('post_type', 'post') === 'post';
```

which would essentially do the same thing (assume that no `post_type` request param means that `post_type` is equal to `posts`), so I can make that change.

Reply:
> mmmm... not seeing that in WP_Posts_List_Table nor in WP_List_Table, where is that happening?

I'm not looking at those.  I'm looking at `wp-admin/edit.php` (and where the `$typenow` global is set in `wp-admin/admin.php`).